### PR TITLE
3 add cicd

### PR DESCRIPTION
**Pull request by mikhasd**
Implemented basic CI/CD pipeline that runs the test, build and deploy into prod when merged to main.

### CODE REVIEW

File: lib/aws-github-oidc.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { Stack, StackProps, CfnOutput} from 'aws-cdk-lib';
+import * as iam from 'aws-cdk-lib/aws-iam';
```
Comment (lucacucchetti) [inline lib/aws-github-oidc.ts:2]: Can we import like `import { Role } from 'aws-cdk-lib/aws-iam';` instead?

File: tsconfig.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
     "lib": [
       "es2018"
     ],
+    "outDir": "./dist",
```
Comment (lucacucchetti) [inline tsconfig.json:8]: Nice

File: package.json [hunk 3]
```diff
@@ -9,6 +9,8 @@
     "watch": "tsc -w",
     "test": "jest",
     "cdk": "cdk",
+    "deploy:oidp": "cdk deploy AwsGithubOidc",
```
Comment (lucacucchetti) [inline package.json:12]: oidc?
```suggestion
    "deploy:oidc": "cdk deploy AwsGithubOidc",
```

**Review by lucacucchetti — COMMENTED** [review]
Nice change! I see there are Actions configured in the repo for this branch, was that done manually?

Will merging this update them automatically or do we need to update those Actions manually?

File: lib/aws-github-oidc.ts (hunk 1 above)
Comment (mikhasd) [reply lib/aws-github-oidc.ts:2]: Sure. Are there any coding standards we should follow?

File: package.json (hunk 3 above)
Comment (mikhasd) [reply package.json:12]: That's a matter of taste: the intention I wanted to express with the command name was "deploy the **O**pen**ID** **C**onnect **P**rovider" which is often referred as OIDP.

Using `deploy:oidc` sounded in my buthead something like `deploy:http` (given that OIDC is a protocol).

Again, that's a matter of taste (I think).

@juancruzlepore any thoughts?

Comment (juancruzlepore) [reply package.json:12]: Oh I looked up the acronym and didn't find it before, but what you say makes sense to me; I am happy with keeping OIDP or even OIDCP

**Review by juancruzlepore — APPROVED** [review]
lgtm

**Pull request merged**